src/containers/DataSet/index.js

```diff
@@ -296,7 +292,7 @@ class DataSetReady extends React.Component {
 
   handleSubmit = async () => {
     if (!this.state.hasToken) {
-      const token = await (await fetch('/token/')).json();
+      const token = await (await asyncFetch('/token/')).json();
```

Comment:
I think we should hold off on changing this one until we're sure what it's doing and how to test it.